cdc_rsync_server/server_socket.h

```diff
@@ -50,11 +50,7 @@ class ServerSocket : public Socket {
                        size_t* bytes_received) override;
 
  private:
-  // Listening socket file descriptor (where new connections are accepted).
-  int listen_sockfd_;
-
-  // Connection socket file descriptor (where data is sent to/received from).
-  int conn_sockfd_;
+  std::unique_ptr<struct ServerSocketInfo> socket_info_;
```

Comment:
Why use a `unique_ptr`? This could just have been a plain instance.

Reply:
Then the platform specific implementation of ServerSocketInfo, including some Windows header info, would leak into the .h file. We'd have to include windows.h (in worst case) or typedef SOCKET (ugly).